solutions/usaco-860.mdx

```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
 
 </JavaSection>
 
+<PySection>
+```python
+import sys
+from collections import deque
+
+sys.stdin = open("mooyomooyo.in", "r")
+sys.stdout = open("mooyomooyo.out", "w")
+
+input = sys.stdin.readline
+
+# possible directions to visit all connected regions
+directions = [
+	(1, 0), 
+	(0, 1), 
+	(-1, 0), 
+	(0, -1)
+]
+
+a, b = map(int, input().split())
+
+def gravity(graph, s, coors):
+
+	for i in range(s // 2):
+		for j in range(5):
+			graph[i][j], graph[j][i] = graph[j][i], graph[i][j]
+    
+	for y, x in coors:
+		graph[y].pop(x)
+		graph[y].append(0)
+    
+	for i in range(5):
+		for j in range(s // 2):
+			graph[i][j], graph[j][i] = graph[j][i], graph[i][j]
+    
+	return graph
+
+# bfs to mark all connected regions
+def bfs(x, y, graph, visited):
```

Comment:
does this need to be in a function as well?

Reply:
yes. kinda.

Reply:
ok then give a better name than `bfs` and also add type annotations (other functions too)

Reply:
what should i name it

Reply:
`mark_connected` or something